File: frontend/src/shared/store/assistant.ts
export type AssistantPromptMode = "full" | "minimal" | "none";
export type AssistantCallMode = "auto" | "custom";
export type AssistantLocaleMode = "auto" | "manual";

export interface AssistantSoul {
  coreTruths?: string[];
  boundaries?: string[];
  rules?: string[];
  vibe?: string;
  continuity?: string;
}

export interface AssistantIdentity {
  name?: string;
  creature?: string;
  emoji?: string;
  role?: string;
  soul?: AssistantSoul;
}

export interface AssistantAvatarAssetRef {
  path?: string;
  displayName?: string;
  source?: string;
  assetId?: string;
  meta?: Record<string, string>;
}

export type Assistant3DAvatar = AssistantAvatarAssetRef;
export type Assistant3DMotion = AssistantAvatarAssetRef;

export interface AssistantAvatar {
  avatar3d?: AssistantAvatarAssetRef;
  motion?: AssistantAvatarAssetRef;
}

export interface UserLocale {
  mode?: AssistantLocaleMode | string;
  value?: string;
  current?: string;
}

export interface UserExtraField {
  key?: string;
  value?: string;
}

export interface AssistantUser {
  name?: string;
  preferredAddress?: string;
  pronouns?: string;
  notes?: string;
  language?: UserLocale;
  timezone?: UserLocale;
  location?: UserLocale;
  extra?: UserExtraField[];
}

export interface ModelConfig {
  inherit?: boolean;
  primary?: string;
  fallbacks?: string[];
  stream?: boolean;
  temperature?: number;
  maxTokens?: number;
}

export interface AssistantModel {
  agent: ModelConfig;
  image: ModelConfig;
  embedding: ModelConfig;
}

export interface AssistantToolItem {
  id?: string;
  enabled: boolean;
}

export interface AssistantTools {
  items?: AssistantToolItem[];
}

export interface AssistantSkills {
  mode?: "on" | "off" | string;
  maxSkillsInPrompt?: number;
  maxPromptChars?: number;
}

export interface CallToolsConfig {
  mode?: AssistantCallMode | string;
  allowList?: string[];
  denyList?: string[];
}

export interface CallSkillsConfig {
  mode?: AssistantCallMode | string;
  allowList?: string[];
}

export interface AssistantCall {
  tools: CallToolsConfig;
  skills: CallSkillsConfig;
}

export interface AssistantMemory {
  enabled: boolean;
}

export interface AssistantReadiness {
  ready: boolean;
  missing?: string[];
}

export interface Assistant {
  id: string;
  builtin: boolean;
  deletable: boolean;
  identity: AssistantIdentity;
  avatar: AssistantAvatar;
  user: AssistantUser;
  model: AssistantModel;
  tools: AssistantTools;
  skills: AssistantSkills;
  call: AssistantCall;
  memory: AssistantMemory;
  readiness: AssistantReadiness;
  enabled: boolean;
  isDefault: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface Assistant3DAvatarAsset {
  kind: "3davatar" | "vrma" | string;
  path: string;
  name: string;
  displayName?: string;
  updatedAt?: string;
  source?: string;
  assetId?: string;
}

export interface ImportAssistant3DAvatarFromPathRequest {
  kind: "3davatar" | "vrma" | string;
  path: string;
}

export interface ReadAssistant3DAvatarSourceRequest {
  kind: "3davatar" | "vrma" | string;
  path: string;
}

export interface ReadAssistant3DAvatarSourceResponse {
  contentBase64: string;
  mime: string;
  fileName: string;
  sizeBytes: number;
}

export interface DeleteAssistantAvatarAssetRequest {
  kind: "3davatar" | "vrma" | string;
  path: string;
}

export interface UpdateAssistantAvatarAssetRequest {
  kind: "3davatar" | "vrma" | string;
  path: string;
  displayName?: string;
}

export interface CreateAssistantRequest {
  identity: AssistantIdentity;
  avatar: AssistantAvatar;
  user: AssistantUser;
  model: AssistantModel;
  tools: AssistantTools;
  skills: AssistantSkills;
  call: AssistantCall;
  memory: AssistantMemory;
  enabled?: boolean;
  isDefault?: boolean;
}

export interface UpdateAssistantRequest {
  id: string;
  identity?: AssistantIdentity;
  avatar?: AssistantAvatar;
  user?: AssistantUser;
  model?: AssistantModel;
  tools?: AssistantTools;
  skills?: AssistantSkills;
  call?: AssistantCall;
  memory?: AssistantMemory;
  enabled?: boolean;
  isDefault?: boolean;
}

export interface DeleteAssistantRequest {
  id: string;
}

export interface SetDefaultAssistantRequest {
  id: string;
}

export interface AssistantMemorySummary {
  summary: string;
}

export interface AssistantProfileOptions {
  roles: string[];
  defaultRole?: string;
  vibes: string[];
  defaultVibe?: string;
}

File: internal/domain/assistant/assistant.go
package assistant

import (
	"strings"
	"time"
)

type Assistant struct {
	ID        string
	Builtin   bool
	Deletable bool
	Identity  AssistantIdentity
	Avatar    AssistantAvatar
	User      AssistantUser
	Model     AssistantModel
	Tools     AssistantTools
	Skills    AssistantSkills
	Call      AssistantCall
	Memory    AssistantMemory
	Enabled   bool
	IsDefault bool
	CreatedAt time.Time
	UpdatedAt time.Time
}

type AssistantParams struct {
	ID        string
	Builtin   *bool
	Deletable *bool
	Identity  AssistantIdentity
	Avatar    AssistantAvatar
	User      AssistantUser
	Model     AssistantModel
	Tools     AssistantTools
	Skills    AssistantSkills
	Call      AssistantCall
	Memory    AssistantMemory
	Enabled   *bool
	IsDefault *bool
	CreatedAt *time.Time
	UpdatedAt *time.Time
}

func NewAssistant(params AssistantParams) (Assistant, error) {
	id := strings.TrimSpace(params.ID)
	if id == "" {
		return Assistant{}, ErrInvalidAssistant
	}
	builtin := false
	if params.Builtin != nil {
		builtin = *params.Builtin
	}
	deletable := true
	if params.Deletable != nil {
		deletable = *params.Deletable
	}
	enabled := true
	if params.Enabled != nil {
		enabled = *params.Enabled
	}
	isDefault := false
	if params.IsDefault != nil {
		isDefault = *params.IsDefault
	}
	createdAt := time.Now()
	updatedAt := createdAt
	if params.CreatedAt != nil {
		createdAt = *params.CreatedAt
	}
	if params.UpdatedAt != nil {
		updatedAt = *params.UpdatedAt
	}
	identity := normalizeAssistantIdentity(params.Identity)
	avatar := normalizeAssistantAvatar(params.Avatar)
	user := normalizeAssistantUser(params.User)
	model := normalizeAssistantModel(params.Model)
	tools := normalizeAssistantTools(params.Tools)
	skills := normalizeAssistantSkills(params.Skills)
	call := normalizeAssistantCall(params.Call)
	memory := normalizeAssistantMemory(params.Memory)

	return Assistant{
		ID:        id,
		Builtin:   builtin,
		Deletable: deletable,
		Identity:  identity,
		Avatar:    avatar,
		User:      user,
		Model:     model,
		Tools:     tools,
		Skills:    skills,
		Call:      call,
		Memory:    memory,
		Enabled:   enabled,
		IsDefault: isDefault,
		CreatedAt: createdAt,
		UpdatedAt: updatedAt,
	}, nil
}
